libraries/stdlib/stdlib_defs.mtlx

```diff
@@ -2494,6 +2494,16 @@
     <output name="out" type="stringarray" defaultinput="in1" />
   </nodedef>
 
+  <!--
+    Node: <trianglewave>
+    Moving between 0.0 and the Scale based on the input value.
+  -->
+  <nodedef name="ND_trianglewave_float" node="trianglewave" nodegroup="math">
+    <input name="input" type="float" value="0" />
+    <input name="scale" type="float" value="0" />
+    <output name="out" type="float" />
+  </nodedef>
```

Comment:
This proposal is starting to come together, @Cinifreak, and I wanted to run a few design suggestions by you and @crydalch for consideration:

1. Should this node have a `freq` input, in the spirit of Chris's recent `checkerboard` node, making it straightforward for an artist to change the frequency of the output triangle wave?
2. Should the default value of the `scale` input be 1 instead of 0, so that the default output of the node is wave-like instead of flat?
3. Perhaps the main input of the node should be named `in` for consistency with other nodes in MaterialX?

Reply:
for point 2 and 3, I totally agree, about the first point i don't see a cause for freq input. It makes sense in checkerboard node. but for this node. it's too simple to have other than scale parameter.

Reply:
@Cinifreak @crydalch Should the `scale` input perhaps be named `amplitude`, in the spirit of this being a node for a wave function?  This would match the precedent of the `noise2d` node, for example, which has an `amplitude` input for its overall scale:

https://github.com/AcademySoftwareFoundation/MaterialX/blob/99ebdf51a42104e032dda16a1bcbf1e582dcea45/documents/Specification/inprog_v1.39/MaterialX.Specification.md?plain=1#L751

Here I'm just making sure we're considering both past and future nodes that have similar properties to `trianglewave`, including other wave-like nodes that we may add to MaterialX in the future.

Reply:
Okay, you have a point. I did the change.